src/components/panel/panel.js

```diff
@@ -329,21 +332,25 @@ MdPanelService.prototype.create = function(opt_config) {
  * A reference to a created panel. This reference contains a unique id for the panel,
  * along with properties/functions used to control the panel.
  *
- * @param {!angular.$injector} $injector
  * @param {!Object} config
+ * @param {!angular.$injector} $injector
  * @final @constructor
  */
-function MdPanelReference(config,$q) {
+function MdPanelReference(config, $injector) {
   // Injected variables.
-  this._$q = $q;
+  this._$q = $injector.get('$q');
+  this._$mdCompiler = $injector.get('$mdCompiler');
 
   // Public variables.
-  this.config = config;
-  this.id     = 'panel_' + this.id;
+  this.id = config.id;
   this.isOpen = false;
 
   // Private variables.
+  this._config = config;
   this._openPromise;
+  this._openReject;
+  this._closePromise;
+  this._panelEl;
```

Comment:
Add JsDoc for these properties where it's not completely obvious (e.g., is `panelEl` an `angular.JQLite` or an `Element`?)